app/questionnaire/navigation.py

```diff
@@ -50,8 +50,20 @@ def build_navigation(self, current_group_id, current_group_instance):
                 else:
                     logger.debug("building navigation", group_id=group['id'])
                     navigation.append(self._build_single_navigation(group, current_group_id, first_location))
+
+        if self._all_sections_completed(navigation):
```

Comment:
There is currently no concept of toggling the visibility of items in the Navigation. Items are either added or not. Whilst I agree that it would probably be more efficient to build the navigation item and toggle its visibility rather than adding the summary/confirmation link to the end, I think it should be looked at in another PR, since it's a larger bit of work to re-work the navigation to work based on the routing path.